### PR TITLE
User name attribute name configure for aad b2c

### PR DESCRIPTION
**Pull request by wujack778**
Configure attribute  `user-name-attribute-name`  for aad-b2c
[task link](https://dev.azure.com/azure-spring-integration/spring-integration-private/_sprints/taskboard/spring-integration-private%20Team/spring-integration-private/182?workitem=410)
[issue link](https://github.com/Azure/azure-sdk-for-java/issues/15313)

### CODE REVIEW

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/b2c/AADB2CProperties.java [hunk 1]
```diff
@@ -68,6 +68,11 @@
 
     private Map<String, Object> authenticateAdditionalParameters;
 
+    /**
+     * User name attribute name,eg "name"
+     */
+    private String UserNameAttributeName;
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/b2c/AADB2CProperties.java:74]: Please update related README and CHANGELOG.

Comment (chenrujun) [reply sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/b2c/AADB2CProperties.java:74]: The variable name should started by lower case.

Comment (wujack778) [reply sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/b2c/AADB2CProperties.java:74]: all right

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-b2c-oidc/README.md [hunk 2]
```diff
@@ -47,6 +47,7 @@ azure:
       client-secret: ${your-client-secret}
       reply-url: ${your-reply-url} # should be absolute url.
       logout-success-url: ${your-logout-success-url}
+      user-name-attribute-name: name
```
Comment (backwind1233) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-b2c-oidc/README.md:50]: image
I think we need to change the task link to issue url

**Review by backwind1233 — COMMENTED** [review]
need change

**Comment by wujack778** [comment]
> change task link to issue link

done

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-b2c-oidc/src/main/java/com/azure/spring/sample/aad/b2c/controller/WebController.java [hunk 3]
```diff
@@ -18,6 +18,7 @@ private void initializeModel(Model model, OAuth2AuthenticationToken token) {
 
             model.addAttribute("grant_type", user.getAuthorities());
             model.addAllAttributes(user.getAttributes());
+            model.addAttribute("name",user.getName());
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-b2c-oidc/src/main/java/com/azure/spring/sample/aad/b2c/controller/WebController.java:21]: Please add space in between.

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/azure-spring-boot-starter-active-directory-b2c/README.md [hunk 4]
```diff
@@ -116,6 +117,7 @@ application to use the Spring annotations and classes to protect the web app.
    | `azure.activedirectory.b2c.reply-url` | Contains one of the **Reply URL** from your application that you completed earlier. |
    | `azure.activedirectory.b2c.logout-success-url` | Specify the URL when your application logout successfully. |
    | `azure.activedirectory.b2c.user-flows` | Contains the name of the user flows that you completed earlier.
+   | `azure.activedirectory.b2c.user-name-attribute-name` | Contains the attribute name of the user name .
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-starter-active-directory-b2c/README.md:120]: How about we put this table under the section of `Key Concept`, just like what the aad starter does https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/spring/azure-spring-boot-starter-active-directory#configurable-properties.

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/azure-spring-boot-starter-active-directory-b2c/README.md (hunk 4 above)
Comment (wujack778) [reply sdk/spring/azure-spring-boot-starter-active-directory-b2c/README.md]: already done

**Review by saragluna — APPROVED** [review]

**Pull request merged**